src/content/reference/react/useOptimistic.md

```diff
@@ -143,3 +148,67 @@ export async function deliverMessage(message) {
 ```
 
 </Sandpack>
+
+---
+
+### Optimistically update the UI while waiting for a network request {/*optimistically-update-while-waiting-for-network*/}
+
+<Sandpack>
+
+```js like-button.js active
+import { startTransition, useOptimistic } from "react";
+
+export function LikeButton({ likes, updateLikes, disabled }) {
+  const [optimisticLikes, addOptimisticLike] = useOptimistic(
```

Comment:
Also the example for me is not updating the like count from 0->1

![CleanShot 2023-12-15 at 09 45 56](https://github.com/reactjs/react.dev/assets/1309636/c4773613-5cc1-4976-9b84-85cd8fc8102f)

re: https://react-ac6nnqh06-fbopensource.vercel.app/reference/react/useOptimistic#noun-labs-1201738-(2)

Reply:
@lunaleaps One way to fix this example would be to start a transition with `useTransition` hook instead of using the standalone `startTransition` function. This way you'll be able to:
- replace the `disabled` prop with the transition's pending state,
- make the arrow function (scope) passed to `startTransition` async and await `updateLikes` inside (crucial).

It fixes the issue while slightly simplifying the code, here's the preview:
https://codesandbox.io/p/sandbox/gallant-nova-8w3s2h